M 9/ejemplos/chat.py

```diff
@@ -0,0 +1,125 @@
+    </head>
+    <body class="bg-gray-100" onload="solicitarNombreUsuario()">
+        <div class="flex flex-row h-screen">
+            <div class="bg-gray-800 text-white w-1/4 p-4">
+                <h2 class="text-xl mb-2">Usuarias</h2>
+                <ul id="usuarios" class="overflow-y-auto"></ul>
+            </div>
+            <div class="flex flex-col w-3/4 p-6">
+                <div class="flex-1 bg-white overflow-y-auto p-4 mb-4 rounded shadow-md">
+                    <ul id="mensajes"></ul>
+                </div>
+                <div class="flex">
+                    <input type="text" id="entradaMensaje" class="border p-2 flex-grow" placeholder="Escribe tu mensaje aquí..." onkeypress="if(event.keyCode == 13) { enviarMensaje(); }">
+                    <button onclick="enviarMensaje()" class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Enviar</button>
+                </div>
+            </div>
+        </div>
+        <script>
+            var ws;
+            var mensajes = document.getElementById('mensajes');
+            var usuarios = document.getElementById('usuarios');
+            var entradaMensaje = document.getElementById('entradaMensaje');
+            var nombreUsuario;
+            function solicitarNombreUsuario() {
+                nombreUsuario = prompt("Por favor, ingresa tu nombre:");
+                if (nombreUsuario) {
+                    ws = new WebSocket("ws://" + window.location.hostname + ":8000/ws?nombre_usuario=" + encodeURIComponent(nombreUsuario));
+                    configurarWebSocket();
+                }
+            }
+
+            function configurarWebSocket() {
+                ws.onmessage = function(evento) {
+                    var datosParseados = JSON.parse(evento.data);
+                    if (datosParseados.tipo === 'mensaje') {
+                        mensajeActual = JSON.parse(datosParseados.mensaje)
+                        mensajes.innerHTML += '<li class="p-2 border-b"><strong>' + datosParseados.usuario + ':</strong> ' + mensajeActual.mensaje + '</li>';
+                    } else if (datosParseados.tipo === 'lista_usuarios') {
+                        usuarios.innerHTML = '';
+                        datosParseados.usuarios.forEach(function(usuario) {
+                            usuarios.innerHTML += '<li class="p-2 border-b">' + usuario + '</li>';
+                        });
+                    }
+                };
+            }
+
+            function enviarMensaje() {
+                var mensaje = entradaMensaje.value;
+                ws.send(JSON.stringify({tipo: 'mensaje', mensaje: mensaje}));
+                entradaMensaje.value = '';
+            }
+        </script>
+    </body>
+</html>
+"""
+
+@app.get("/")
+async def obtener():
+    return HTMLResponse(html)
+
+# Clase para manejar las conexiones de WebSocket
+class AdministradorConexiones:
+    def __init__(self):
+        self.conexiones_activas: Dict[str, WebSocket] = {}
```

Comment:
```suggestion
        self.conexiones_activas: dict[str, WebSocket] = {}
```

En Python 3.9+ ya no hay que importar el `Dict` :tada: